src/metabase/query_processor/middleware/process_userland_query.clj

```diff
@@ -132,26 +132,31 @@
 (defn- query-execution-info
   "Return the info for the QueryExecution entry for this `query`."
   {:arglists '([query])}
-  [{{:keys [executed-by query-hash context action-id card-id dashboard-id pulse-id]} :info
-    database-id                                                                      :database
-    query-type                                                                       :type
-    :as                                                                              query}]
+  [{{:keys [executed-by query-hash context action-id card-id dashboard-id pulse-id original-query]} :info
+    database-id                                                                                     :database
+    query-type                                                                                      :type
+    :as                                                                                             query}]
   {:pre [(bytes? query-hash)]}
-  {:database_id       database-id
-   :executor_id       executed-by
-   :action_id         action-id
-   :card_id           card-id
-   :dashboard_id      dashboard-id
-   :pulse_id          pulse-id
-   :context           context
-   :hash              query-hash
-   :native            (= (keyword query-type) :native)
-   :json_query        (cond-> (dissoc query :info)
-                        (empty? (:parameters query)) (dissoc :parameters))
-   :started_at        (t/zoned-date-time)
-   :running_time      0
-   :result_rows       0
-   :start_time_millis (System/currentTimeMillis)})
+  (let [json-query (if original-query
+                     (-> (read-string original-query)
+                         (dissoc :info)
+                         (assoc :was-pivot true))
+                     (cond-> (dissoc query :info)
+                       (empty? (:parameters query)) (dissoc :parameters)))]
```

Comment:
So, I don't think this is a great approach yet.
Why am I doing this at all?

The api/dataset namespace (what we use to run unsaved questions), when loading a Pivot Question, FIRST does a POST to `dataset/pivot`. 

The response from that endpoint carries the rows in what I'll call a 'raw pivoted' form. Eg, it will have one column called 'pivot-grouping' and will have rows near the end of the results with nulls and numbers (which are the row and grand totals).

The response ALSO carries with it a query! That's pretty nice, right? It's this exact query that gets passed to 

`POST dataset/csv`. 

The problem here is that it's NOT the correct query. What query is it? It's the last run query from the `qp.pivot` machine. Details don't matter at this stage, it's just important to know that our pivot implementation runs multiple queries to construct the raw pivoted results, and the final query does not necessarily contain all of the necessary columns/aggregations/metadata, whatever.

What is needed is to pass the same query that gets passed into the `POST dataset/pivot` endpoint at the start of this whole thing, and the only way I could think to keep it is to keep it secret (keep it safe) 🧙🏼‍♂️in the :info of the query, and tag it with :was-pivoted so that, finally, the dataset query :export-format endpoint can switch to the correct qp runner when doing a CSV download.

🤯This has twisted my brain a bit, but hopefully that helps clarify the _why_ of this json-query business.